mbed_targets/mbed_targets.py

```diff
@@ -84,22 +88,71 @@ class DatabaseMode(Enum):
 
     OFFLINE = 0
     ONLINE = 1
+    AUTO = 2
+
+
+def get_target(product_code: str, mode: DatabaseMode) -> MbedTarget:
+    """Get an MbedTarget by its product code.
+
+    Args:
+        product_code: the product code to look up in the database.
+        mode: a DatabaseMode enum field.
+    """
+    if mode == DatabaseMode.OFFLINE:
```

Comment:
We can either make CodeClimate ignore this or extract 3 private functions. I guess this is a bit on the long side, but simple enough that I'm indifferent.

Reply:
I've refactored it slightly in the next commit. If CodeClimate still has problems with the complexity we can tell it to ignore this.